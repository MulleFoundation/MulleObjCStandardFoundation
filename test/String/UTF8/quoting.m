//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"



static unsigned char    hoehoe[] =
{
   'T', 'A', 'B', ':', '\t',   '\n',
   '\\', '1',     ':', '\001', '\n',
   '\\', 'f',     ':', '\f',   '\n',
   '\\', 'e',     ':', '\e',   '\n',
   'D', 'E', 'L', ':',  127,   '\n',
   '?', '\n',
   '\xe2', '\x98', '\x84', '\xef', '\xb8', '\x8f', '\xe2', '\x98',
   '\x83', '\xef', '\xb8', '\x8f', '\xf0', '\x9f', '\x91', '\x8d',
   '\xf0', '\x9f', '\x8f', '\xbe'
};


int   main( int argc, const char * argv[])
{
   NSString   *s;
   NSString   *quoted;
   NSString   *unquoted;

   s = [NSString mulleStringWithUTF8Characters:hoehoe
                                        length:sizeof( hoehoe)];
   if( ! s)
      return( 1);

   quoted   = [s mulleQuotedString];
   unquoted = [quoted mulleUnquotedString];
   printf( "C        : %s\n", hoehoe);
   printf( "Original : %s\n", [s UTF8String]);
   printf( "Quoted   : %s\n", [quoted UTF8String]);
   printf( "Unquoted : %s\n", [unquoted UTF8String]);

   return( [unquoted isEqualToString:s] == YES ? 0 : -1);
}
