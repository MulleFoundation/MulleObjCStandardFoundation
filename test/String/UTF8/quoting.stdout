C        : TAB:	
\1:
\f:
\e:
DEL:
?
☄️☃️👍🏾
Original : TAB:	
\1:
\f:
\e:
DEL:
?
☄️☃️👍🏾
Quoted   : "TAB:\t\n\\1:\001\n\\f:\f\n\\e:\e\nDEL:\177\n\?\n☄️☃️👍🏾"
Unquoted : TAB:	
\1:
\f:
\e:
DEL:
?
☄️☃️👍🏾
